bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/model/SomfyTahomaDevice.java

```diff
@@ -67,4 +68,18 @@ public SomfyTahomaDeviceDefinition getDefinition() {
     public List<SomfyTahomaState> getAttributes() {
         return attributes;
     }
+
+    public @Nullable SomfyTahomaState getState(String stateName, @Nullable Integer stateType) {
+        return getState(states, stateName, stateType);
+    }
+
+    public static @Nullable SomfyTahomaState getState(List<SomfyTahomaState> states, String stateName,
+            @Nullable Integer stateType) {
+        for (SomfyTahomaState state : states) {
+            if (stateName.equals(state.getName()) && (stateType == null || stateType == state.getType())) {
+                return state;
+            }
+        }
+        return null;
+    }
```

Comment:
This is a model class. Please keep is as a simple POJO

Reply:
Ok, I moved the method into `SomfyTahomaBaseThingHandler`.